nvm.sh

```diff
@@ -191,6 +191,65 @@ nvm_find_up() {
   nvm_echo "${path}"
 }
 
+nvm_find_package_json() {
+  local dir
+  dir="$(nvm_find_up 'package.json')"
+  if [ -e "${dir}/package.json" ]; then
+    nvm_echo "${dir}/package.json"
+  fi
+}
+
+nvm_get_node_from_pkg_json() {
+  local engines_object=''
+  local open_brackets=0
```

Comment:
because ksh doesn't have `local`, the `local foo` and `foo=0` pieces have to be on separate lines.